mypyc/genops.py

```diff
@@ -1975,11 +1977,44 @@ def decompose_union_helper(self,
         self.activate_block(exit_block)
         return result
 
-    def isinstance(self, obj: Value, rtype: RInstance, line: int) -> Value:
-        class_ir = rtype.class_ir
-        fullname = '%s.%s' % (class_ir.module_name, class_ir.name)
-        type_obj = self.load_native_type_object(fullname)
-        return self.primitive_op(fast_isinstance_op, [obj, type_obj], line)
+    def isinstance_helper(self, obj: Value, class_irs: List[ClassIR], line: int) -> Value:
+        """Fast path for isinstance() with native classes."""
+        if not class_irs:
+            return self.primitive_op(false_op, [], line)
+        ret = self.isinstance_single(obj, class_irs[0], line)
+        for class_ir in class_irs[1:]:
+            ret = self.primitive_op(fast_or_op,
+                                    [ret, self.isinstance_single(obj, class_ir, line)],
+                                    line)
+        return ret
+
+    def isinstance_single(self, obj: Value, class_ir: ClassIR, line: int) -> Value:
```

Comment:
Maybe rename to `isinstance_native`? The key feature seems to be that this works with native classes.

Reply:
👍